src/main/java/MongoDBDAO.java

```diff
@@ -1,139 +1,140 @@
+        //For Development:
         MongoCollection<Document> collection = MongoConnection.getInstance("AgricircleDB").getDatabase().getCollection("Weatherstation1");
 
         StringBuilder returnString = new StringBuilder();
-        ObjectId date = DateToObjectId(stringDate);
-        FindIterable<Document> iterable = collection.find(gte("_id" , date));
-        System.out.println(iterable);
-        if (iterable != null) {
-            for (Document document : iterable) {
-                returnString.append(document.toJson());
-            }
+        ObjectId date;
+        try{
+             date = DateToObjectId(stringDate);
         }
-        if(returnString.length() == 0)
+        catch(WrongDateFormatException wdfe)
+        {
+            return wdfe.getMessage();
+        }
+
+        FindIterable<Document> iterable = collection.find(gte("_id" , date)).
+                projection(fields(include(
+                        "payload_fields.avg_wind_speed",
+                        "payload_fields.solar_radiation",
+                        "payload_fields.outside_temperature",
+                        "payload_fields.outside_humidity",
+                        "payload_fields.barometer_data",
+                        "payload_fields.rain_rate"),
+                        excludeId()));
+
+        if(contentType!= null && contentType.equals("netcdf"))
         {
-            return "We couldn't find any data with the given criteria";
+            //TODO Convert JSON to netcdf and return it.
+            return "Not Yet Implemented. Use contenttype = json";
+        }
+        else
+        {
+            if (iterable != null) {
+                returnString.append("{\"dataobjects\":[");
+                for (Document document : iterable) {
+                    returnString.append(document.toJson()).append(",");
+                }
+                returnString.delete(returnString.length() - 1, returnString.length());
+                returnString.append("]}");
+            }
         }
         return returnString.toString();
     }
 
-    private static String DateToHexString(int year, int month, int day)
-    {
+    private static String DateToHexString(int year, int month, int day) throws WrongDateFormatException {
         final Calendar calendar = Calendar.getInstance();
         calendar.set(year, month-1, day , 0 , 0 , 0);
 
         calendar.set(Calendar.MILLISECOND, 0);
 
         if(calendar.getTime().after(new Date()))
         {
-            throw new InputMismatchException("Date should be earlier than current date");
+            throw new WrongDateFormatException("Date should be earlier than current date");
         }
         long time = (calendar.getTimeInMillis() / 1000);
         return Long.toHexString(time) + "0000000000000000";
     }
 
-    private ObjectId DateToObjectId(String stringDate)
-    {
+    private static ObjectId DateToObjectId(String stringDate) throws WrongDateFormatException {
```

Comment:
Why are you making ObjectIds from dates?

Reply:
Tomasz wanted the endpoint to return data between a given date, and the current date.
One way to filter is taking advantage of the generated _id in Mongo.
The _id consists of a field containing a timestamp. Therefore i take the date given as part of the path, convert it to a ObjectID and use it as a filter.